scripts/solhint-custom/index.js

```diff
@@ -0,0 +1,82 @@
+const path = require('path');
+const minimatch = require('minimatch');
+
+// Files matching these patterns will be ignored unless a rule has `static global = true`
+const ignore = ['contracts/mocks/**/*', 'test/**/*'];
+
+class Base {
+  constructor(reporter, config, source, fileName) {
+    this.reporter = reporter;
+    this.ignored = this.constructor.global || ignore.some(p => minimatch(path.normalize(fileName), p));
+    this.ruleId = this.constructor.ruleId;
+    if (this.ruleId === undefined) {
+      throw Error('missing ruleId static property');
+    }
+  }
+
+  error(node, message) {
+    if (!this.ignored) {
+      this.reporter.error(node, this.ruleId, message);
+    }
+  }
+}
+
+module.exports = [
+  class extends Base {
+    static ruleId = 'interface-names';
+
+    ContractDefinition(node) {
+      if (node.kind === 'interface' && !/^I[A-Z]/.test(node.name)) {
+        this.error(node, 'Interface names should have a capital I prefix');
+      }
+    }
+  },
+
+  class extends Base {
+    static ruleId = 'private-variables';
+
+    VariableDeclaration(node) {
+      const constantOrImmutable = node.isDeclaredConst || node.isImmutable;
+      if (node.isStateVar && !constantOrImmutable && node.visibility !== 'private') {
+        this.error(node, 'State variables must be private');
+      }
+    }
+  },
+
+  class extends Base {
+    static ruleId = 'underscore-prefix';
+
+    VariableDeclaration(node) {
+      const constantOrImmutable = node.isDeclaredConst || node.isImmutable;
+      if (node.isStateVar && !constantOrImmutable && node.visibility === 'private') {
+        if (!/^_/.test(node.name)) {
+          this.error(node, 'Private variables must be prefixed with underscore');
+        }
+      }
```

Comment:
```suggestion
      }
      if (node.isStateVar && constantOrImmutable) {
        if (/^_/.test(node.name)) {
          this.error(node, 'Constant and immutable variables should not be prefixed with underscore');
        }
      }
```

Reply:
I agree, but this causes the rule to fail in a few places.

I added the rule but commented it out, and I'd propose to do another PR next week fixing those things. Does this sound ok?

Reply:
So we will have 2 follow up PRs that :
- enable the rule 
- fix the code
?

That sounds ok

Reply:
Yeah or just one PR doing the two.

Reply:
Maybe the rule should also check that the name is capitalized.

Reply:
That should be already handled by the casing rules in `solhint.config.js`, specifically `const-name-snakecase`.